### ISSUE TITLE
Throws JsResultException when I upload an image file.

### ISSUE THREAD

**Issue by 3tty0n** [gilbertw1/slack-scala-client#23]
Hi Bryan.

When I upload an image file, such as the following exception is thrown.

```
[ERROR] [06/13/2016 15:34:47.481] [slack-akka.actor.default-dispatcher-3] [akka://slack/user/$a] [SlackRtmClient] Error parsing text frame
play.api.libs.json.JsResultException: JsResultException(errors:List((/file/size,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/pretty_type,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/mode,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/url,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/user,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/is_external,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/filetype,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/created,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/title,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/editable,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/timestamp,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/external_type,List(ValidationError(List(error.path.missing),WrappedArray()))), (/file/mimetype,List(ValidationError(List(error.path.missing),WrappedArray())))))
	at play.api.libs.json.JsReadable$$anonfun$2.apply(JsReadable.scala:20)
	at play.api.libs.json.JsReadable$$anonfun$2.apply(JsReadable.scala:20)
	at play.api.libs.json.JsResult$class.fold(JsResult.scala:73)
	at play.api.libs.json.JsError.fold(JsResult.scala:13)
	at play.api.libs.json.JsReadable$class.as(JsReadable.scala:18)
	at play.api.libs.json.JsObject.as(JsValue.scala:76)
	at slack.models.package$$anon$1.reads(package.scala:233)
	at play.api.libs.json.JsValue$class.validate(JsValue.scala:18)
	at play.api.libs.json.JsObject.validate(JsValue.scala:76)
	at play.api.libs.json.JsReadable$class.as(JsReadable.scala:18)
	at play.api.libs.json.JsObject.as(JsValue.scala:76)
	at slack.rtm.SlackRtmConnectionActor$$anonfun$receive$1.applyOrElse(SlackRtmClient.scala:103)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:480)
	at slack.rtm.SlackRtmConnectionActor.aroundReceive(SlackRtmClient.scala:87)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

**Comment by gilbertw1** [comment]
Hi, thanks for reporting this! I'll take a look ASAP.

**Comment by andyh2** [comment]
I see a similar exception when uploading a CSV file with SlackApiClient.uploadFile. For what it's worth, the file does in fact upload to the desired channel but the exception still fires -- must be on handling the response?

`[info] Error uploading file to Slack: play.api.libs.json.JsResultException: JsResultException(errors:List((/url,List(ValidationError(List(error.path.missing),WrappedArray())))))`

**Comment by 3tty0n** [comment]
> must be related to handling the response?

I think so.

I handle this exception in my application as below, but `play.api.libs.json.JsResultException` is thrown as usual.

```Scala
try {
  apiClient.uploadFile(file = new File(s"image.png"), channels = Some(Seq(generalChanId)))
} catch {
  e: play.api.libs.json.JsResultException =>
    println(e.getMessage())
}
```

So, must be handling this exception in SlackApiClient ?

**Comment by manojkakarla** [comment]
This is after the changes made by slack, details here https://api.slack.com/changelog/2016-04-26-file-events-in-the-rtm-api

**Comment by gilbertw1** [comment]
Hey guys, this is fixed in the latest release: 0.1.7. Thanks for reporting!

**Comment by 3tty0n** [comment]
Great!! Thanks!!
